crates/mongodb-agent-common/src/query/arguments/json_to_bson.rs

```diff
@@ -0,0 +1,424 @@
+        let object_type_name = "scalar_test".to_owned();
+        let object_type = ObjectType {
+            fields: BTreeMap::from([
+                ObjectField::new("double", Type::Scalar(BsonScalarType::Double)),
+                ObjectField::new("int", Type::Scalar(BsonScalarType::Int)),
+                ObjectField::new("long", Type::Scalar(BsonScalarType::Long)),
+                ObjectField::new("decimal", Type::Scalar(BsonScalarType::Decimal)),
+                ObjectField::new("string", Type::Scalar(BsonScalarType::String)),
+                ObjectField::new("date", Type::Scalar(BsonScalarType::Date)),
+                ObjectField::new("timestamp", Type::Scalar(BsonScalarType::Timestamp)),
+                ObjectField::new("binData", Type::Scalar(BsonScalarType::BinData)),
+                ObjectField::new("objectId", Type::Scalar(BsonScalarType::ObjectId)),
+                ObjectField::new("bool", Type::Scalar(BsonScalarType::Bool)),
+                ObjectField::new("null", Type::Scalar(BsonScalarType::Null)),
+                ObjectField::new("undefined", Type::Scalar(BsonScalarType::Undefined)),
+                ObjectField::new("regex", Type::Scalar(BsonScalarType::Regex)),
+                ObjectField::new("javascript", Type::Scalar(BsonScalarType::Javascript)),
+                ObjectField::new(
+                    "javascriptWithScope",
+                    Type::Scalar(BsonScalarType::JavascriptWithScope),
+                ),
+                ObjectField::new("minKey", Type::Scalar(BsonScalarType::MinKey)),
+                ObjectField::new("maxKey", Type::Scalar(BsonScalarType::MaxKey)),
+                ObjectField::new("symbol", Type::Scalar(BsonScalarType::Symbol)),
+            ]),
+            description: Default::default(),
+        };
+
+        let input = json!({
+            "double": 3.14159,
+            "int": 3,
+            "long": 3,
+            "decimal": "3.14159",
+            "string": "hello",
+            "date": "2024-03-22T00:59:01Z",
+            "timestamp": { "t": 1565545664, "i": 1 },
+            "binData": {
+                "base64": "EEEBEIEIERA=",
+                "subType": "00"
+            },
+            "objectId": "e7c8f79873814cbae1f8d84c",
+            "bool": true,
+            "null": null,
+            "undefined": null,
+            "regex": { "pattern": "^fo+$", "options": "i" },
+            "javascript": "console.log('hello, world!')",
+            "javascriptWithScope": {
+                "$code": "console.log('hello, ', name)",
+                "$scope": { "name": "you!" },
+            },
+            "minKey": {},
+            "maxKey": {},
+            "symbol": "a_symbol",
+        });
+
+        let expected = bson::doc! {
+            "double": Bson::Double(3.14159),
+            "int": Bson::Int32(3),
+            "long": Bson::Int64(3),
+            "decimal": Bson::Decimal128(bson::Decimal128::from_str("3.14159")?),
```

Comment:
Clippy really doesn't like you trying to impersonate pi. I found that with one of my test cases too 😂 

https://github.com/hasura/ndc-mongodb/actions/runs/8446549353/job/23135506300?pr=16#step:6:163